### PR TITLE
feat: project api sort for merged applications

### PR DESCRIPTION
**Pull request by Yakitrak**
## Describe your changes

Ticket: https://pins-ds.atlassian.net/browse/ASB-2282

PR adds API level sort for merged (BO + NI) projects. Sorting is done on API level after merged and mapped. Like other merged functionality, it is temporarily and will be removed after we have migrated all cases to BO. 

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/applications-service-api/src/utils/sort-applications.merge.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const sortApplications = (applications, sort = '+ProjectName') => {
+	const direction = sort.startsWith('-') ? 'desc' : 'asc';
+	const key = sort.replace(/^[+-]/, '');
+
+	switch (key) {
+		case 'ProjectName':
+		case 'PromoterName':
+			return applications.sort(sortByStringKeyInObject(key, direction));
+		case 'DateOfDCOSubmission':
+		case 'ConfirmedDateOfDecision':
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/src/utils/sort-applications.merge.js:16]: What is the behaviour in the absence of this field in the project?  For example if only 2 cases had a ConfirmedDateOfDecision.  

I don't think there is a scenario in the ticket for this but would be good to have a secondary order of ProjectName.

Comment (Yakitrak) [reply packages/applications-service-api/src/utils/sort-applications.merge.js:21]: Thank you the suggestion - I've added now!

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**